content/plugin/log/writing.mdx

```diff
@@ -0,0 +1,137 @@
+
+```go
+tflog.Trace(ctx, "executing API request", "url", "https://www.example.com/my/endpoint", "method", "POST")
+```
+You can use types other than strings as values, too:
+  TOD: What are teh acceptable types? Are there any we don't support that might surprise folks?
+
+```go
+tflog.Trace(ctx, "executing API request", "url", "https://www.example.com/my/endpoint", "method", "POST", "size", 200, "authenticated", true, "headers", map[string][]string{"content-type": []string{"application/json"}})
+```
+
+## Logger Context
+
+Loggers are transported using the `context.Context` variable, so injecting a variable into a logger returns a new `context.Context` variable containing the modified logger. Any calls to `tflog` using that logger will then implicitly include that variable. For example, the `terraform-plugin-go` library uses this method to assign a unique request ID to every RPC. This request ID that is included in all log output, allowing you to filter for logs associated with a specific request.
+
+```go
+newContext := tflog.With(ctx, "url", "https://www.example.com/my/endpoint")
+
+// won't include the `url` variable
+tflog.Debug(ctx, "this won't have the url variable attached")
+
+// will include the `url` variable
+tflog.Debug(newContext, "this will have the url variable attached")
+```
+
+## Subsystems
+
+You may want to create a new subsystem for sections of code that are large, complex, or have distinct functionality. You can configure each subsystem to be included or excluded from log output using environment variables. For example, you may want to create a subsystem for logs that relate to the API client. These logs can be useful, but you may want the ability to turn them off when when debugging an unrelated issue.
+
+### Create Subsystems
+
+To create a new subsystem, pass context and the subsystem name to the `NewSubsystem()` method.
+
+```go
+// my-subsystem is the name of the subsystem
+// it will be attached to all your logs
+ctx = tflog.NewSubsystem(ctx, "my-subsystem")
+```
+
+Optionally, you can also specify a [logging level](#logging-levels) for the subsystem.
+
+```go
+ctx = tflog.NewSubsystem(ctx, "my-subsystem", hclog.Debug)
+```
+
+You can also configure an environment variable to control the logging level instead of hardcoding it into the subsystem.
+
+```go
+// read the level from TF_LOG_MYPROVIDER_CLIENT
+ctx = tflog.NewSubsystem(ctx, "my-subsystem", tflog.WithLevelFromEnv("TF_LOG_MYPROVIDER_CLIENT"))
+```
+
+### Use Subsystems
+
+Use the `Subsystem*` functions to associate log output with a specific subsystem.
+
+```go
+tflog.SubsystemDebug(ctx, "my-subsystem", "writing to a subsystem", "meaning_of_life", 42)
+```
+
+Just like with `tflog.With`, it's important that the `Subsystem*` functions get passed the context that is returned from `tflog.NewSubsystem`; the modified logger is stored in the context.
```

Comment:
@bflad I don't really understand what this means. And should we show an example?

Reply:
YES - thank you